### PR TITLE
skip binary and empty files

### PR DESCRIPTION
**Pull request by dfrankland**
https://linear.app/trunk/issue/TRUNK-4736/trunk-toolbox-failed-to-build-line-views

### CODE REVIEW

**Comment by dfrankland** [comment]
closes https://github.com/trunk-io/trunk/pull/5816

**Review by zaycev — APPROVED** [review]

File: src/rules/pls_no_land.rs [hunk 1]
```diff
@@ -25,25 +26,31 @@ pub fn pls_no_land(paths: &HashSet<PathBuf>) -> anyhow::Result<Vec<diagnostic::D
     Ok(ret)
 }
 
-type LinesView = Vec<String>;
+fn pls_no_land_impl(path: &PathBuf) -> anyhow::Result<Vec<diagnostic::Diagnostic>> {
```
Comment (zaycev) [inline src/rules/pls_no_land.rs:29]: I think it's better to accept the reader. The file may not necessary be on file system.

Comment (dfrankland) [reply src/rules/pls_no_land.rs:29]: This was the signature before, and I believe we get a list of files to lint by the Trunk CLI—is there something else you were thinking of?

Comment (zaycev) [reply src/rules/pls_no_land.rs:29]: Oh, sorry I miss understood. I see that you've combined two functions together here. Nvm then

**Pull request merged**